timApp/modules/cs/js/util/util.ts

```diff
@@ -13,7 +13,12 @@ export function countWords(str: string): number {
     if (s.length === 1 && s[0].length === 0) {
         return 0;
     }
-    return s.length;
+    // ignore punctuation characters that appear on their own, as the grammar/syntax in some languages (notably, French)
+    // dictates whitespace before some punctuation, like question/exclamation marks etc.
+    const ans = s.filter((e: string) => !e.match(/^\p{P}/gu));
```

Comment:
Mielestäni toimii hyvin. Testasin, ja ainakin esimerkiksi `!testi` ei laskettu sanaksi.

Mietin, pitäisikö sellainen typo sallia, eli siis laskea se sanaksi jos joku nyt sattuu kirjoittamaan yhteen merkin. Parempi antaa hyvä arvio

Eli olisiko regex jotain tyyliä 

```suggestion
    const ans = s.filter((e: string) => !e.match(/^\p{P}+$/gu));
```

Reply:
Ehdinkin jo korjailla vähän eri lailla ennen kuin huomasin kommentin.
Korjasin tuon muotoon

    const ans = s.filter((e) => e.match(/\p{P}?\w/u));

joka tekee saman. Se on ehkä helpommin luettava tuossa kontekstissa? Makuasia toki :)

Reply:
...josta regexin saa edelleen yksinkertaistettua muotoon /\w/u  :)

Reply:
Joo, tuo onkin nyt hyvä.